packages/client/src/createClient.ts

```diff
@@ -82,33 +85,74 @@ function createFutureClientPlus(
 
 // Once we migrate everyone off of using the deprecated parts of `Client` we can rename this to `createClient`.
 // For now, its a way to use JUST the new client
-export function createFutureClient(
-  metadata: MinimalClientParams["metadata"],
+export function createFutureClient<V extends VersionString<any, any, any>>(
+  metadata: SatisfiesSemver<V, MaxOsdkVersion> extends true
+    ? MinimalClientMetadata<V>
+    : MinimalClientMetadata<MaxOsdkVersion> & {
+      [ErrorMessage]:
+        `Your SDK requires a semver compatible version with ${V}. You have ${MaxOsdkVersion}. Update your package.json`;
+    },
   stack: string,
   tokenProvider: () => Promise<string> | string,
   ontologyCachingOptions: OntologyCachingOptions = {},
   fetchFn: typeof globalThis.fetch = fetch,
-): FutureClient {
+): SatisfiesSemver<V, MaxOsdkVersion> extends true ? FutureClient
+  : never
+{
   // When `createFutureClient` gets renamed to `createClient`, we
   // should inline this call as its no longer needed to be separate.
   return createFutureClientPlus(
-    metadata,
+    metadata as MinimalClientMetadata<V>,
     stack,
     tokenProvider,
     ontologyCachingOptions,
     fetchFn,
-  )[1];
+  )[1] as any;
+}
+
+function isOntologyMetadata(
+  ontology: OntologyDefinition<any> | OntologyMetadata<any>,
+): ontology is MinimalClientMetadata<any> {
+  return "ontologyRid" in ontology || "ontologyApiName" in ontology;
 }
 
-export function createClient<O extends OntologyDefinition<any>>(
-  ontology: O,
+const MaxOsdkVersion = "0.14.0";
```

Comment:
Is this something we'd need to keep updating?

Reply:
Sadly.